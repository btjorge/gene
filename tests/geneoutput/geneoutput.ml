let prefix = "../progs/"

let progs = [ "max02"; "max03"; "max04"; "max05"; "max06"; "max3" ]

let symbinterp file =
  let command = "../../bin/gene.exe " ^ prefix ^ file in
  print_endline command;
  if Sys.command command <> 0
  then failwith "error";
  ()

let () =
  List.iter symbinterp progs
