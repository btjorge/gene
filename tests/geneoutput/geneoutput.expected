../../bin/gene.exe ../progs/max02
The following test parameters are not used by the code (see the generic test):
_x0, _x1
Generalized test:
assertEquals (max (_x0, _x1), _x2)
assertEquals (max (5, 2), 5)  genericity: 0
../../bin/gene.exe ../progs/max03
The following test parameters are not used by the code (see the generic test):
_x1
Generalized test:
assertEquals (max (_x0, _x1), _x2)
assertEquals (max (5, 2), 5)  genericity: 1
../../bin/gene.exe ../progs/max04
The following test parameters are not used by the code (see the generic test):
_x0
Generalized test:
assertEquals (max (_x0, _x1), _x2)
assertEquals (max (5, 2), 5)  genericity: 1
../../bin/gene.exe ../progs/max05
assertEquals (max (5, 2), 5)  genericity: 1
../../bin/gene.exe ../progs/max06
assertEquals (max (5, 2), 5)  genericity: 1
../../bin/gene.exe ../progs/max3
assertEquals (max3 (1, 3, 2), 3)  genericity: 1
